### PR TITLE
feature/#CU-a0jmgg-Document-Get-booking-info-by-trip-id-API-Design

### PR DESCRIPTION
**Pull request by lotusmp87**
## Changed
- Added API for getting booking detail by trip id

### CODE REVIEW

**Comment by lotusmp87** [comment]
Task linked: [CU-a0jmgg Document Get booking info by trip id API Design](https://app.clickup.com/t/a0jmgg)

File: tn_api_pricing_booking_spec.yml [hunk 1]
```diff
@@ -878,35 +953,33 @@ components:
...
-                        currency: 
-                            description: Currency in which the booking was made
-                            type: string
-                            example: CAD
-                        payment:       
-                            type: boolean
-                            description: Notes whether the booking included payment information. 
-                            example: true
+
+    passenger:
```
Comment (mcclymont-k) [inline tn_api_pricing_booking_spec.yml:1113]: Based on the design we want a few more passenger details like email, phone, passport # - I dont remember which ones we decided on though.....
image

Comment (lotusmp87) [reply tn_api_pricing_booking_spec.yml:1113]: Done

**Comment by lotusmp87** [comment]
@mcclymont-k  this is ready for another review

**Review by mcclymont-k — APPROVED** [review]

**Review by andrescollart — CHANGES_REQUESTED** [review]
We need to return the flight details beyond just the pricing side so we can display flight departure times, etc. That's not currently scoped out in this response.

Request seems fine :)

**Comment by trip-ninja-inc** [comment]
Changes requested:

All in the response
1.  itinerary/segments/flight_details should be an array of flight_details objects, not just one.
2. passengers (email and phone number should be optional, except for Primary Passenger). Passport_number and expiry date are also optional I think.

Written by Andres

**Comment by lotusmp87** [comment]
@andrescollart this is ready for a re-review

**Comment by mcclymont-k** [comment]
We need status in the base of the response here: 
Example in the response below 

image

File: tn_api_pricing_booking_spec.yml [hunk 2]
```diff
@@ -858,55 +1039,134 @@ components:
...
+        gender:
+          description: Gender
+          type: string
+          example: "M"
+          enum: [F, M]
+        passport_number:
+          description: Passport Number *Optional*
+          type: string
+          example: "KDS3468"
+        passport_expiry_date:
```
Comment (Iceman997) [inline tn_api_pricing_booking_spec.yml]:
```suggestion
        passport_expiration:
```

File: tn_api_pricing_booking_spec.yml [hunk 3]
```diff
@@ -858,55 +1039,134 @@ components:
...
+          description: Passport Expiry Date *Optional*
+          type: string
+          format: date
+          example: "2021-11-21"
+        nationality:
```
Comment (Iceman997) [inline tn_api_pricing_booking_spec.yml]:
```suggestion
        passport_country:
```

File: tn_api_pricing_booking_spec.yml [hunk 4]
```diff
@@ -858,55 +1039,134 @@ components:
...
+          description: Nationality *Optional*
+          type: string
+          example: "Canada"
+        type:
```
Comment (Iceman997) [inline tn_api_pricing_booking_spec.yml]:
```suggestion
        passenger_type:
```

So these match the booking request object

File: tn_api_pricing_booking_spec.yml [hunk 5]
```diff
@@ -858,55 +1039,134 @@ components:
...
+          enum: [ADT, STU, CHD, INF]
+        email:
+          description: passenger email address *Primary passenger only*
+          type: string
+          example: "andres.collar@tripninja.io"
+        phone_number:
+          description: passenger phone number *Primary passenger only*
+          type: string
+          example: "9022222222"
+
```
Comment (Iceman997) [inline tn_api_pricing_booking_spec.yml:1174]: Also missing 'date_of_birth'

Comment (mcclymont-k) [reply tn_api_pricing_booking_spec.yml:1174]: Yup it should match this object:
image

PS - im fixing it within the DB table on my trip ID PR

Comment (lotusmp87) [reply tn_api_pricing_booking_spec.yml:1174]: Added

**Comment by lotusmp87** [comment]
@Iceman997  @mcclymont-k  this is ready for re-review

**Review by andrescollart — APPROVED** [review]

**Review by Iceman997 — APPROVED** [review]

**Pull request merged**